lusidtools/cocoon/instruments.py

```diff
@@ -326,6 +337,7 @@ def get_unique_identifiers(api_factory: lusid.utilities.ApiClientFactory):
     )
 
     names = []
+    figis = []
 
     for response in responses:
         name = np.NaN
```

Comment:
You may want to add figi = np.NaN here, otherwise you will use the figi from the previous succesful search in all subsequent searches until you find a new figi.